### PR TITLE
Ox trust/issues/#815

### PR DESCRIPTION
**Pull request by shekhar16**
Show Modality for otp, mobile, passport, fido devices.

### CODE REVIEW

File: model/src/main/java/org/gluu/oxtrust/model/GluuCustomPerson.java [hunk 1]
```diff
@@ -53,8 +53,30 @@
 
     @LdapAttribute(name = "oxExternalUid")
     private List<String> oxExternalUid;
+    
+    @LdapAttribute(name = "oxOTPDevices")
+    private String oxOTPDevices;
```
Comment (yurem) [inline model/src/main/java/org/gluu/oxtrust/model/GluuCustomPerson.java]: We should change this to:
```
    @LdapJsonObject
    @LdapAttribute(name = "oxOTPDevices")
    private OTPDevice oxOTPDevices;
```

File: model/src/main/java/org/gluu/oxtrust/model/GluuCustomPerson.java [hunk 2]
```diff
@@ -53,8 +53,30 @@
 
     @LdapAttribute(name = "oxExternalUid")
     private List<String> oxExternalUid;
```
Comment (yurem) [inline model/src/main/java/org/gluu/oxtrust/model/GluuCustomPerson.java]: Use `@LdapJsonObject`

File: server/src/main/java/org/gluu/oxtrust/action/UpdatePersonAction.java [hunk 3]
```diff
@@ -255,18 +267,55 @@ public String update() {
...
 	                deviceDataMap.add(gluuDeviceDataBean); 
 				}
-			}
+			}			
 			
-			if(oxexternal != null && oxexternal.size()>0){
-				for(String oxexternalStr : oxexternal){
-					String [] args = oxexternalStr.split(":");
-					GluuDeviceDataBean gluuDeviceDataBean= new GluuDeviceDataBean();
+			if (oxOTPDevices != null && !oxOTPDevices.trim().equals("")) {
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/action/UpdatePersonAction.java]: Who and where add this list o user entry?

File: server/src/main/java/org/gluu/oxtrust/action/UpdatePersonAction.java [hunk 4]
```diff
@@ -255,18 +267,55 @@ public String update() {
...
+						gluuDeviceDataBean.setModality("OTP Device");
+						gluuDeviceDataBean.setId(device.getId());
+						Timestamp stamp = new Timestamp(Long.valueOf(device.getAddedOn()).longValue());
+						gluuDeviceDataBean.setCreationDate(stamp.toGMTString());
+						deviceDataMap.add(gluuDeviceDataBean);
+					}
+				}
+			}
+
+			if (oxMobileDevices != null && !oxMobileDevices.trim().equals("")) {
```
Comment (yurem) [inline server/src/main/java/org/gluu/oxtrust/action/UpdatePersonAction.java]: Who and where add this list o user entry?

File: model/src/main/java/org/gluu/oxtrust/model/GluuCustomPerson.java (hunk 2 above)
Comment (shekhar16) [reply model/src/main/java/org/gluu/oxtrust/model/GluuCustomPerson.java]: its not an json string.

**Pull request merged**